src/components/Swap/Toolbar/Caption.tsx

```diff
@@ -59,24 +76,27 @@ function Caption({ icon: Icon = AlertTriangle, caption, color = 'secondary', too
   )
 }
 
-function GasEstimate({ gasUseEstimateUSD }: GasEstimateProps) {
+function GasEstimate({ gasUseEstimateUSD, trade }: GasEstimateProps) {
   const isWideWidget = useIsWideWidget()
   // TODO(just-toby): use formatCurrencyAmount from conedison
   const displayEstimate = !gasUseEstimateUSD
     ? '-'
     : formatCurrencyAmount({ amount: gasUseEstimateUSD, isUsdPrice: true })
+  if (gasUseEstimateUSD == null) {
+    return null
+  }
   return (
     <CaptionRow gap={0.25}>
-      {isWideWidget ? (
-        <>
+      <>
+        {trade ? (
```

Comment:
what happens here if its a narrow size widget? it should display the gas fee in a tooltip right?

Reply:
i was displaying the gas fee text in a tooltip, but the design actually prefers to show that information under the routing diagram, so i am replacing my label with the RoutingDiagram. I still need to update the RoutingDiagram component to match the design

Reply:
oo got it, can you add a TODO in the code or add a "followup item" in the PR description just so we know for posterity that this PR needs the quick follow to be feature complete :3